### ISSUE TITLE
'https:' protocol passed to 'caw' module instead of 'https'

### ISSUE THREAD

**Issue by sheakelly** [kevva/download#85]
````
var protocol = url.parse(get.url).protocol;
var agent = caw(this.opts.proxy, {protocol: protocol});
````
In the above code the protocol var gets the value 'https:' for secure urls. This is passed to the 'caw' module that expects 'https'. The result is that agent uses 'httpOverHttp' instead of the desired 'httpsOverHttp'. We end up with a http call to github that is infinately redirects to https.

**Comment by sheakelly** [comment]
Sorry logged against wrong project

**Comment by sheakelly** [comment]
Reopening

**Comment by shinnn** [comment]
Confirmed.

**Comment by shinnn** [comment]
Fixed in v4.4.2.

**Comment by sheakelly** [comment]
I have tested your change in version v4.4.2. The `protocol.slice(9, -1)` is wiping out the protocol. Always getting httpOverHttp now.

**Comment by shinnn** [comment]
Oh, I can't believe I did that. Fixed in https://github.com/kevva/download/commit/5d93aac42454d24cc30445fb172bbd38ace125b2 and released as v4.4.3.
